eras/shelley/impl/testlib/Test/Cardano/Ledger/Shelley/ImpTest.hs

```diff
@@ -885,11 +1020,28 @@ impNativeScriptKeyPairs tx = do
   keyPairs <- mapM (impSatisfyNativeScript curAddrWits) nativeScripts
   pure . mconcat $ catMaybes keyPairs
 
+fixupTxOuts :: ShelleyEraImp era => Tx era -> ImpTestM era (Tx era)
+fixupTxOuts tx = do
+  pp <- getsNES $ nesEsL . curPParamsEpochStateL
+  let
+    txOuts = tx ^. bodyTxL . outputsTxBodyL
+  fixedUpTxOuts <- forM txOuts $ \txOut -> do
+    let txOut' = ensureMinCoinTxOut pp txOut
+    when (txOut /= txOut') $ do
+      logEntry $
+        "Fixed up the amount in the TxOut from "
+          ++ show (txOut ^. coinTxOutL)
+          ++ " to "
+          ++ show (txOut' ^. coinTxOutL)
+    pure txOut'
+  pure $ tx & bodyTxL . outputsTxBodyL .~ fixedUpTxOuts
+
 fixupFees ::
   (ShelleyEraImp era, HasCallStack) =>
   Tx era ->
   ImpTestM era (Tx era)
-fixupFees tx = impAnn "fixupFees" $ do
+fixupFees txOriginal = impAnn "fixupFees" $ do
+  tx <- fixupTxOuts txOriginal
```

Comment:
This fixup is one of the biggest changes in this PR, which we need because genesis file forces minutxo value for all outputs. For this reason none of those `sendCoinTo addr (Coin 10)` will work, simply because the amount being sent is now too small.

Thinking a bit more about this function after your review I decided to change it a bit, namely amount will be fixed up only if it is originally set to `0`. This way it will be a lot less surprising when writing tests.